planetscale/certs.go

```diff
@@ -57,10 +59,20 @@ func (c *certificatesService) Create(ctx context.Context, r *CreateCertificateRe
 		CommonName: cn,
 	}
 
+	var signatureAlgorithm x509.SignatureAlgorithm
+	switch r.PrivateKey.(type) {
+	case *rsa.PrivateKey:
+		signatureAlgorithm = x509.SHA256WithRSA
+	case *ecdsa.PrivateKey:
+		signatureAlgorithm = x509.ECDSAWithSHA256
+	default:
+		return nil, fmt.Errorf("unsupported key type, only supports ECDSA & RSA private keys")
```

Comment:
It would be nice to include the unsupported key type:

```suggestion
		return nil, fmt.Errorf("unsupported key type: %T, only supports ECDSA & RSA private keys", r.PrivateKey)
```

Reply:
Done and also made some further cleanups.